src/org/rascalmpl/library/lang/csv/IO.java

```diff
@@ -387,7 +387,7 @@ public IValue visitSet(Type type) throws RuntimeException {
                 }
                 @Override
                 public IValue visitSourceLocation(Type type) throws RuntimeException {
-                    return values.sourceLocation(URIUtil.invalidURI());
+                    return URIUtil.unknownLocation();
```

Comment:
is unknown location something different than invalid uri?  (as in, why rename it from invalid to unknown?)

Reply:
Yes. "Invalid" means "wrong" , while "unknown" can be a valid state meaning that it is uninitialized yet or not knowable. That's not wrong.